app/services/foreman_ansible/playbooks_importer.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module ForemanAnsible
+  # Imports playbooks from smart proxy
+  class PlaybooksImporter
+    include ::ForemanAnsible::ProxyAPI
+    delegate :playbooks, :playbooks_names, :to => :proxy_api
+
+    def initialize(proxy = nil)
+      @ansible_proxy = proxy
+    end
+
+    def import_playbooks(playbooks_names)
+      playbooks = playbooks(playbooks_names)
+      result = { created: {}, updated: {}, deleted: {} }
+      playbooks.each do |playbook|
+        parsed_playbook = parse_playbook playbook
+        job_template_id = JobTemplate.where(name: parsed_playbook[:name]).pick(:id)
+        if job_template_id.present?
+          updated = update_job_template(job_template_id, parsed_playbook)
+          result[:updated].merge!(updated) unless updated.nil?
+        else
+          result[:created].merge!(create_job_template(parsed_playbook))
+        end
+      end
+      result[:deleted] = delete_job_templates(playbooks_names)
+      result
+    end
+
+    def parse_playbook(playbook)
+      content = playbook['playbooks_content'].join('')
+      {
+        name: playbook['name'],
+        playbook_content: add_metadata(playbook['name']) + content,
+        vars: get_vars(content)
+      }
+    end
+
+    def add_metadata(playbook_name)
+      [
+        '<%#',
+        "  name: #{playbook_name}",
+        '  snippet: false',
+        '  job_category: Ansible Playbook - Imported',
+        '  provider_type: Ansible',
+        '  kind: job_template',
+        '  model: JobTemplate',
+        "%>\n"
+      ].join('')
+    end
+
+    def get_vars(playbook_content)
+      YAML.safe_load(playbook_content).map { |play| play['vars'] }
+    end
+
+    def create_job_template(playbook)
+      job_template = JobTemplate.create(name: playbook[:name], template: playbook[:playbook_content], job_category: 'Ansible Playbook - Imported', provider_type: 'Ansible')
+      # TODO: Add support for creating template inputs
+      job_template.save
+      { job_template.id => job_template.name }
+    end
+
+    def update_job_template(job_template_id, playbook)
+      # TODO: Add support for updating template inputs
+      inputs = []
+      job_template = JobTemplate.find(job_template_id)
+      should_update = !playbook[:playbook_content].eql?(job_template.template)
+      job_template.template = playbook[:playbook_content] if should_update
+      { job_template.id => job_template.name } unless inputs.empty? && !should_update
+    end
+
+    def delete_job_templates(playbooks_names)
+      JobTemplate.where(job_category: 'Ansible Playbook - Imported').where.not(name: playbooks_names).destroy_all.map { |template| { template.id => template.name } } unless playbooks_names.empty?
```

Comment:
This sounds dangerous. Do we want to delete all templates that are not explicitly imported?

What if I import some from one proxy and then I import another from different proxy, the second import would delete all the playbooks from the first import, wouldn't it?

Reply:
Yes, you are right. 
It was the logic a while ago, but I agree it has some troubles. 
When should we delete job templates?

Reply:
I believe there should eighter be a special input for deleted playbooks, or there should be a separate method for deleting them. Optionally, you can just not implement the delete at all (users are able to delete the playbooks from the job_templates directly, so the need to implement it on import is not quite there :))